docs/content/example.js

```diff
@@ -0,0 +1,15 @@
+/**
```

Comment:
I would prefer `example.ts` (since we do run typescript, just without strict checking of types)

Reply:
Right, there might 2 reasons for this for now:

- there seems to be an issue with JupyterLite showing the `.ts` file as base64 encoded (probably a lite issue, to be investigated)
- arguably a JavaScript example is the simplest form of extension we can expose to folks new to web dev. Similar to the idea of using the JS cookiecutter as a simple version of the extension cookiecutter: https://github.com/jupyterlab/extension-cookiecutter-js

Happy to revisit this and make TS the default, but could also be done as a follow-up.

Reply:
> there seems to be an issue with JupyterLite showing the .ts file as base64 encoded (probably a lite issue, to be investigated)

can't know'em all: need to add to [`extra_file_types`](https://jupyterlite.readthedocs.io/en/latest/howto/content/files.html#customizing-mime-types).

Reply:
Ah right I thought we had this one by default already.

Reply:
Switched back to `example.ts`, although still thinking that a JS example could be relevant to folks new to the JS / TS ecosystem. We can get feedback about this and iterate as needed.

Reply:
I see your point. Well, maybe we could have both? I guess it does not matter as much for lite until we have LSP support. But once we have LSP in lite (which should be doable here), the TS/JS server gives better hints in TS mode (obviouly).

Reply:
> LSP in lite (which should be doable here),

Oh, indeed. That's been low-prof on my backburner for a while. As, I've mentioned, I think an in-browser `json-language-server` would have huge bang for the buck as being shipped in-core _Lab_, not just lite.

Getting to full-on pygls-based servers like jedi-language-server would be the next target... but the general case, often requiring `subprocess.Popen` or equivalent, is going to be real hard. Indeed, I reckon my whole "LSP-over-comms" rant would actually be even _more_ relevant in lite, as at present, you'd need a second emcsripten computer with a _big_ PROXYFS to run a blocking language server.

> a JS example could be relevant to folks new to the JS / TS ecosystem

Yerp: I can't usually in good conscience suggest someone write non-TS labextensions, even to start with. The hill is already quite steep from "i got nothing" or "jquery-and-pray" to full-on _you can't do **anything** without Generics, Promises, Signals, Interfaces, Tokens_ and doing so by "save-reload-refresh-ooops-hard-refresh" is maddening.

For now, I feel we should "push the veal" with a default `.ts` (and _not_ `.tsx`).

It's worth exploring a baseline `.js` _entirely_ annotated and typecheckable (and _checked_) with [JSDoc-style annotations](https://incrementalelm.com/typescript-without-transpilation/) the never actually gets transpiled... which, alas, will _never_ be the case, due to webpack and babel. Being able to learn about `tsconfig.json` interactively is indeed one of the things the plugin playground probably _should_ do.

To that earlier notational point, it's frustrating that the `requires` and `optional` positional types for an `activate`, which one can't really avoid interacting with, are as poor as they are, but I can't imagine how confusing they would be without types. My kingdom for something that required _zero_ types (much less _generics_), but still gave one the full benefits:

```js
const fooPlugin = JupyterFrontEndPlugin({
  provides: IFoo,
  depends: {
    required: {notebooks: INotebookTracker},
    optional: {mainMenu: IMainMenu},
  },
  activate: (app, deps) => {
    deps.notebook.doThing  // always works
    deps.mainMenu.doThing // red squiggly yo-this-might-be-null
    deps.crazyPants.doThing // o no, wat?
  } 
});
```

To the earlier point: all would be right in the world if `.mjs` worked perfectly, and deduplicating module federation somehow worked fully off validateable `package.json` data (again with LSP support), and you could author code directly into your `labextension/dist` without `jupyter lab build`... but we can't really solve that problem here.

Reply:
> which, alas, will never be the case, due to webpack and babel.

Did you see https://github.com/tc39/proposal-type-annotations? Agree on other bits.

Reply:
> which, alas, will never be the case, due to webpack and babel.

Did you see https://github.com/tc39/proposal-type-annotations? Agree on other bits.